### PR TITLE
CMake/common adaptions

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
Needs https://github.com/BlueBrain/Tuvok/pull/10

File: livre/uvf/CMakeLists.txt [hunk 1]
```diff
@@ -8,21 +8,9 @@ if(NOT TUVOK_FOUND)
...
-add_library( LivreUVF SHARED ${LIVRE_UVF_SOURCES} )
-target_link_libraries( LivreUVF PRIVATE ${LIVRE_UVF_TARGET_LIBRARIES} )
-set_target_properties( LivreUVF PROPERTIES VERSION ${VERSION} SOVERSION ${VERSION_ABI} )
-set_target_properties( LivreUVF PROPERTIES COMPILE_FLAGS "${LIVRE_UVF_COMPILE_FLAGS}" )
-if(NOT APPLE)
-  set_target_properties( LivreUVF PROPERTIES LINK_FLAGS "-Wl,--no-undefined" )
-endif()
-install( TARGETS LivreUVF LIBRARY DESTINATION lib )
-install_files(include/livre/UVF FILES ${LIVRE_UVF_HEADERS} )
+common_library( LivreUVF ${LIVREUVF_HEADERS} ${LIVRE_UVF_SOURCES} )
```
Comment (bilgili) [inline livre/uvf/CMakeLists.txt]: set_target_properties( LivreUVF PROPERTIES LINK_FLAGS "-Wl,--no-undefined" )		

This enforces the library to have no unidentified symbols ? Is it covered by the CMake subtree ?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -20,10 +20,6 @@ file(COPY "uvf/mouse_reduced.uvf" DESTINATION ${UVF_DATA_DIR})
 add_definitions(-DUVF_DATA_FILE=\"${UVF_DATA_FILE}\")
 
 # Create and install all the tests
-if(CMAKE_COMPILER_IS_GNUCXX_PURE)
-  set(data-dataSource_LINK_LIBRARIES "-Wl,--no-as-needed")
```
Comment (bilgili) [inline tests/CMakeLists.txt]: ditto ?

File: livre/uvf/CMakeLists.txt (hunk 1 above)
Comment (tribal-tec) [inline livre/uvf/CMakeLists.txt]: Yes, moved to CommonCompiler.cmake. It's applied for all targets now.

**Comment by tribal-tec** [comment]
Updated. Builds now.

**Pull request merged**